Sources/Subprocess/Thread.swift

```diff
@@ -114,20 +115,22 @@ private final class WorkQueue: Sendable {
             pthread_mutex_unlock(mutex)
         }
         #endif
+        if condition(&queue) {
+            #if canImport(WinSDK)
+            SleepConditionVariableCS(self.waitCondition, mutex, INFINITE)
+            #else
+            pthread_cond_wait(self.waitCondition, mutex)
```

Comment:
This doesn't look right. `pthread_cont_wait` and pretty much all other low-level condition variable implementations I have used can wake up spuriously. So usually they need to be in a loop. I.e. `while condition(&queue) {` instead of `if condition(&queue)`